### PR TITLE
fix(experiments): Prevent saving funnel exp with only one goal

### PR DESCRIPTION
**Pull request by danielbachhuber**
## Changes

Prevents saving a Conversion Funnel Experiment when there's only one funnel step goal defined:

CleanShot 2024-11-05 at 10 27 05@2x

CleanShot 2024-11-05 at 10 28 09

## How did you test this code?

I created a new Experiment with two funnel steps as the goal and saved it as a draft. I then clicked "Change goal" from the Experiment page, removed one of the funnel steps, and observed the disabled "Save" button. I then added another funnel step and was able to save the Goals modal successfully.

I also edited an existing Trend experiment and verified I could save the modal with only one goal defined.

### CODE REVIEW

File: frontend/src/scenes/experiments/ExperimentView/Goal.tsx [hunk 1]
```diff
@@ -113,7 +115,10 @@ export function ExperimentGoalModal({ experimentId }: { experimentId: Experiment
                     </LemonButton>
                     <LemonButton
                         disabledReason={
-                            goalInsightDataLoading && 'The insight needs to be loaded before saving the goal'
+                            (goalInsightDataLoading && 'The insight needs to be loaded before saving the goal.') ||
```
Comment (danielbachhuber) [inline frontend/src/scenes/experiments/ExperimentView/Goal.tsx:118]: @jurajmajerik Periods at the end of tooltips, right?

**Comment by danielbachhuber** [comment]
@jurajmajerik One idiosyncrasy I ran into: you can create a new draft Conversion Experiment with only one funnel step:

CleanShot 2024-11-05 at 10 42 38@2x

Once it's created, you can't edit the experiment goal until you define a second funnel step:

CleanShot 2024-11-05 at 10 43 01@2x

Thoughts?

**Comment by jurajmajerik** [comment]
That's fine! We want to remove the goal setting step from the creation form anyway (see item 4. of your onboarding list).

File: frontend/src/scenes/experiments/ExperimentView/Goal.tsx (hunk 1 above)
Comment (jurajmajerik) [reply frontend/src/scenes/experiments/ExperimentView/Goal.tsx:118]: I'm not sure what the convention here should be, fine by me 👍 

Not mentioned in our style guide: https://posthog.com/handbook/growth/marketing/posthog-style-guide

**Review by jurajmajerik — APPROVED** [review]
🚢

**Pull request merged**